egs/wsj/s5/steps/nnet3/lstm/make_configs.py

```diff
@@ -170,17 +188,42 @@ def ParseSpliceString(splice_indexes, label_delay=None):
     prev_layer_output = nodes.AddLdaLayer(config_lines, "L0", prev_layer_output, args.config_dir + '/lda.mat')
 
     for i in range(args.num_lstm_layers):
-        prev_layer_output = nodes.AddLstmLayer(config_lines, "Lstm{0}".format(i+1), prev_layer_output, args.cell_dim,
-                                         args.recurrent_projection_dim, args.non_recurrent_projection_dim,
-                                         args.clipping_threshold, args.norm_based_clipping,
-                                         args.ng_per_element_scale_options, args.ng_affine_options,
-                                         lstm_delay = lstm_delay[i])
+	if len(lstm_delay[i]) == 2: # BLSTM layer case, add both forward and backward
+	    try:
+	        if not (lstm_delay[i][0] < 0 and lstm_delay[i][1] > 0):
+	            raise ValueError('Warning: ' + str(lstm_delay[i]) + ' is not a standard BLSTM mode. There should be a negative delay for the forward, and a postive delay for the backward.')
+	    except ValueError as e:
+		print(str(e))
+            prev_layer_output1 = nodes.AddLstmLayer(config_lines, "Lstm{0}f".format(i+1), prev_layer_output, args.cell_dim,
```

Comment:
We already use the name "f" for forget gates in the LSTM. I would suggest changing the layer names to BLSTM{0}_forward and  BLSTM{0}_backward.